packages/clay-drop-down/src/Menu.tsx

```diff
@@ -0,0 +1,110 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import * as React from 'react';
+import classNames from 'classnames';
+import {Align} from 'metal-position';
+import {Portal} from '@clayui/shared';
+import {useDropdownCloseInteractions} from './hooks';
+
+interface Props extends React.HTMLAttributes<HTMLDivElement> {
+	/**
+	 * Flag to indicate if menu is showing or not.
+	 */
+	active?: boolean;
+
+	/**
+	 * HTML element that the menu should be aligned to
+	 */
+	alignElementRef: React.RefObject<HTMLElement>;
+
+	/**
+	 * Default position of menu element. Values come from `metal-position`.
+	 *
+	 * Align.TopCenter = 0;
+	 * Align.TopRight = 1;
+	 * Align.RightCenter = 2;
+	 * Align.BottomRight = 3;
+	 * Align.BottomCenter = 4;
+	 * Align.BottomLeft = 5;
+	 * Align.LeftCenter = 6;
+	 * Align.TopLeft = 7;
+	 *
+	 * Defaults to BottomLEft
+	 */
+	alignmentPosition?: number;
+
+	/**
+	 * Flag to indicate if menu is displaying a clay-icon on the left.
+	 */
+	hasLeftSymbols?: boolean;
+
+	/**
+	 * Flag to indicate if menu is displaying a clay-icon on the right.
+	 */
+	hasRightSymbols?: boolean;
+
+	/**
+	 * Callback function for when active state changes.
+	 */
+	onSetActive: (val: boolean) => void;
+}
+
+const DropDownMenu = React.forwardRef<HTMLDivElement, Props>((
```

Comment:
@bryceosterhaus we would have to deal with focus control later, but as it is composition we have to think of a way to do that.

Reply:
I have a few ideas how this might be fixed, let me see if I can add something

Reply:
Great! This may help in the creation of Autocomplete, I am waiting for the DataProvider and Dropdown to come in to send the proposal.

Reply:
Wasn't super happy with the solution I came up with. Gonna go ahead and merge for now, and still look into ideas for this.

Reply:
https://github.com/liferay/clay/issues/1987